addons/website_slides/controllers/main.py

```diff
@@ -239,6 +243,22 @@ def _build_channel_domain(self, base_domain, slide_type=None, my=False, **post):
             domain = expression.AND([domain, [('partner_ids', '=', request.env.user.partner_id.id)]])
         return domain
 
+    def _channel_remove_session_answers(self, channel, slide=False):
+        """ Will remove the answers saved in the session for a specific channel / slide. """
+
+        if 'slide_answer_quiz' not in request.session:
+            return
+
+        slides_domain = [('channel_id', '=', channel.id)]
+        if slide:
+            slides_domain = expression.AND([slides_domain, [('id', '=', slide.id)]])
```

Comment:
no need to have channel_id in the domain if slide is set, am I wrong ?

Reply:
It's not absolutely needed no, but it doesn't hurt.